doc: Adds migration example from `mongodbatlas_cluster` to `mongodbatlas_advanced_cluster` (#3058)
mongodb/terraform-provider-mongodbatlas: examples/migrate_cluster_to_advanced_cluster/basic/README.md

> # Basic Migration from `mongodbatlas_cluster` to `mongodbatlas_advanced_cluster`

Reviewer: q: Is there a reason we provide this new guide within the examples section and not in existing https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/cluster-to-advanced-cluster-migration-guide? How will users know when to reference each one?
I personally was thinking of having all the information in a unified place.

Reviewer 2: @AgustinBettati cc @EspenAlbert I think this is to give context for the example, but we're also going to update the guide you mention

> The purpose of this example is to show how a `mongodbatlas_cluster` can be migrated to a `mongodbatlas_advanced_cluster`.

Reviewer 2: ```suggestion
The purpose of this example is to show how a `mongodbatlas_cluster` resource can be migrated to `mongodbatlas_advanced_cluster`.
```

Author: Thanks!

> In this example we use specific files, but the same approach can be applied to configuration with `mongodbatlas_cluster` resource(s).

Reviewer 2: ```suggestion
In this example we use specific files, but the same approach can be applied to any configuration file with `mongodbatlas_cluster` resource(s).
```

Author: Thanks!

The main steps are:

- [Create the `mongodbatlas_cluster`](#create-the-mongodbatlas_cluster) (skip if you already have a configuration with one or more `mongodbatlas_cluster` resources)
- [Use the Atlas CLI Plugin Terraform to create the `mongodbatlas_advanced_cluster` resources](#use-the-atlas-cli-plugin-terraform-to-create-the-mongodbatlas_advanced_cluster-resource)
- [Manual updates to the Terraform configuration](#manual-updates-to-the-terraform-configuration)
- [Perform the Move](#perform-the-move)
  - [Troubleshooting](#troubleshooting)

## Create the `mongodbatlas_cluster`
**Note**: This step is only to demonstrate the migration. If you want to manage a cluster with Terraform, we recommend you use a `mongodbatlas_advanced_cluster` instead.

> This step can be skipped if you already have a configuration file with a `mongodbatlas_cluster` created

Reviewer 2: ```suggestion
This step can be skipped if you already have a configuration file with a `mongodbatlas_cluster` created.
```

Author: Thanks!

> 1. Uncomment the code in [cluster.tf](cluster.tf) and [outputs.tf](outputs.tf) (lines marked with `# BEFORE`)

Reviewer 2: ```suggestion
1. Uncomment the code in [outputs.tf](outputs.tf) (lines marked with `# BEFORE`) and [cluster.tf](cluster.tf)
```

nit: to make it clearer that # BEFORE is only for outputs.tf, in the beginning lookat cluster.tf I was a bit confused about that

Author: Thanks!

> 2. Comment the code in [advanced_cluster](advanced_cluster.tf) and [outputs.tf](outputs.tf) (lines marked with `# AFTER`)

Reviewer 2: ```suggestion
2. Comment the code in [outputs.tf](outputs.tf) (lines marked with `# AFTER`) and [advanced_cluster](advanced_cluster.tf) 
```

Reviewer 3: nit: all the other files have .tf even in the text, so it should be `[advanced_cluster.tf](advanced_cluster.tf)`

Author: Thanks! Added both changes

3. Create a `vars.auto.tfvars` file, for example:
```terraform
project_id = "{PROJECT_ID}" # replace with your project ID, should be similar to 664619d870c247237f4b86a6
cluster_name = "cluster-mig-resource"
instance_size = "M10"
mongo_db_major_version = "8.0"
```
4. Run `terraform init`
5. Run `terraform apply`

## Use the [Atlas CLI Plugin Terraform](https://github.com/mongodb-labs/atlas-cli-plugin-terraform) to create the `mongodbatlas_advanced_cluster` resource

> The CLI Plugin helps you migrate your Terraform configuration file.

Reviewer 3: ```suggestion
The CLI Plugin helps you convert your existing configuration into one using the `mongodbatlas_advanced_cluster` resource. This step is not required but recommended; alternatively you can write the equivalent configuration on your own.
```

1. Ensure you have the Atlas CLI [installed](https://www.mongodb.com/docs/atlas/cli/current/install-atlas-cli/) by running `atlas --version`
   - You should see a multi line output starting with `atlascli version: 1.38.0` (or a later version)
2. Install the Terraform CLI plugin: `atlas plugin install mongodb-labs/atlas-cli-plugin-terraform`
3. Run `atlas terraform` to ensure the plugin was installed correctly
   - You should see a multi line output starting with `Utilities for Terraform's MongoDB Atlas Provider`
4. Run `atlas tf clusterToAdvancedCluster --file {CLUSTER_IN}.tf --output {CLUSTER_OUT}.tf`
   1. For example `cluster.tf` for `{CLUSTER_IN}.tf`
   2. For example `advanced_cluster.tf` for `{CLUSTER_OUT}.tf`

>    3. If your config is not supported, try to update your `mongodbatlas_cluster`, or see [TODO: GUIDE]() for alternative migration approaches

Reviewer 3: what would be the case of not supported? do we already have a page to link?

Author: Added both limitations and a link to our existing migration guide

## Manual updates to the Terraform configuration
1. Ensure all references are updated (see example of updates in [outputs.tf](outputs.tf))
2. Comment out the `mongodbatlas_cluster` in `{CLUSTER_IN}.tf`
3. Add the moved block for each resource migrated in `{CLUSTER_OUT}.tf`
```terraform
moved {
  from = mongodbatlas_cluster.this # change `this` to your specific resource identifier
  to   = mongodbatlas_advanced_cluster.this # change `this` to your specific resource identifier
}
```
- The [moved block](https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#moved-block-syntax) can be kept to record where you have historically moved or renamed an object

## Perform the Move

> 1. Ensure you are using V2 schema: `export MONGODB_ATLAS_ADVANCED_CLUSTER_V2_SCHEMA=true`

Reviewer 2: ```suggestion
1. Ensure you are using V2 schema: `export MONGODB_ATLAS_PREVIEW_PROVIDER_V2_ADVANCED_CLUSTER=true`
```

can already be changed

Author: Thanks!

      ~ backup_enabled                       = false -> true
      ~ connection_strings                   = {
          + private          = (known after apply)
          + private_endpoint = (known after apply)
          + private_srv      = (known after apply)
          ~ standard         = "mongodb://cluster-mig-resource-shard-00-00.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-01.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-02.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-03.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-04.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-05.jciib.mongodb-dev.net:27017,cluster-mig-resource-shard-00-06.jciib.mongodb-dev.net:27017/?ssl=true&authSource=admin&replicaSet=atlas-46mqxd-shard-0" -> (known after apply)
          ~ standard_srv     = "mongodb+srv://cluster-mig-resource.jciib.mongodb-dev.net" -> (known after apply)
        } -> (known after apply)
        name                                 = "cluster-mig-resource"
        # (22 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
- You might see errors like:

> - `Error: Invalid index`: This is due to type changes and can usually be resolved by removing the `[0]` reference.

Reviewer 2: ```suggestion
- `Error: Invalid index`: This is due to type changes and can usually be resolved by removing the `[0]` or `.0` reference.
```

Author: Thanks!
